### ISSUE TITLE
Terminal: provide an action to clear the terminal

### ISSUE THREAD

**Issue by bpasero** [Microsoft/vscode#6603]
Similar to the output panel would be nice to have an action top right to clear the terminal contents.

**Comment by Tyriar** [comment]
This will probably involve sending a particular key sequence to the xterm object.

**Comment by chrmarti** [comment]
The following clears the terminal with two (related) issues:
- There will be no prompt after clearing (like: bash-3.2$ <cursor>)
- Clearing while running something like vim or top will leave the terminal unusable.

	const esc = String.fromCharCode(27);
	this.xterm.write(`${esc}[H${esc}[2J`);

I'd suggest to not implement this and rely on the shell's built-in clearing (like `clear` in bash).

**Comment by Tyriar** [comment]
How about calling into [`eraseInDisplay`](https://github.com/sourcelair/xterm.js/blob/master/src/xterm.js#L3238) with Erase All arguments?

**Comment by chrmarti** [comment]
That's what <esc>[2J does. <esc>H additionally positions the caret in the top left corner. The problem remains that we don't know in what context we clear the terminal and therefore don't know how to properly prompt for input or if we even should clear the terminal in the first place.

**Comment by Tyriar** [comment]
You're right, I still this this should be done but it will need to be done upstream https://github.com/sourcelair/xterm.js/issues/173

**Comment by chrmarti** [comment]
There is a CSI code to lookup the current cursor position. That and a few others could potentially be used to delete all lines except the current one.

**Comment by Tyriar** [comment]
Exposing this in upstream would make things a lot more convenient for consumers, all they will need to do it `xterm.clear()` then. It should also be much safer since `eraseInDisplay` should really not be exposed, also sending key codes will fail if the terminal is currently executing a process.

**Comment by Tyriar** [comment]
Added the API in upstream and implemented the command. The ⌘K keybinding on mac is still blocked by https://github.com/Microsoft/vscode/issues/12187

**Comment by Tyriar** [comment]
Moved the keybinding to https://github.com/Microsoft/vscode/issues/12585 as it's not going to land for Sept.
